src/Command.php

```diff
@@ -232,15 +232,15 @@ public function getCommand()
      */
     public function getExecCommand()
     {
-        if ($this->_execCommand===null) {
-            $command = $this->getCommand();
-            if (!$command) {
-                $this->_error = 'Could not locate any executable command';
-                return false;
-            }
-            $args = $this->getArgs();
-            $this->_execCommand = $args ? $command.' '.$args : $command;
+        $command = $this->getCommand();
+        if (!$command) {
+            $this->_error = 'Could not locate any executable command';
+            return false;
         }
+
+        $args = $this->getArgs();
+        $this->_execCommand = $args ? $command.' '.$args : $command;
+        
         return $this->_execCommand;
```

Comment:
```suggestion
        return $args ? $command.' '.$args : $command;
```

Could you please also remove the `_execCommand` property?

Reply:
As you mentioned before, we need to reset `$args` too after `setCommand`, right?

Reply:
Like I said: I don't want to add this. You simply should not reuse the same instance if you have a completely different comand. If everything is different why not create a new instance?

Reply:
Ok, that understood.